common/common/src/main/java/io/helidon/build/common/FileUtils.java

```diff
@@ -1139,4 +1142,21 @@ public static URL urlOf(Path path) {
             throw new RuntimeException(e);
         }
     }
+
+    /**
+     * Check if file is a submodule.
+     *
+     * @param path tested file path
+     * @return {@code true} if the file is a submodule, {@code false} otherwise
+     */
+    public static boolean isSubModule(Path path) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(path.toFile(), StandardCharsets.UTF_8))) {
```

Comment:
That's too specific to be in FileUtils. You can design a method like `containsLine` instead that takes a `Predicate<String>`.

E.g.
```java
public static boolean containsLine(Path path, Predicate<String> predicate) {
    try (Stream<String> lines = Files.lines(path)){
        return lines.anyMatch(predicate);
    } catch (IOException e) {
        throw new UncheckedIOException(e);
    }
}
```